### PR TITLE
Company view

### PR DESCRIPTION
**Pull request by alfredoct96**
**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Automated tests included? | N
| Related user documentation PR URL |  /
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | /
| BC breaks? | /
| Deprecations? | /

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Added view page for companies.
It includes:
* Basic company's info
* Button to export company's info
* Graph with sum of engagement and points of each company's contact
* Table with pagination of all contacts of company

The request so includes all routes to get all contacts of company and export company info.

I added also methods for 
* view company page
* get sum of engagements and score of each company's user
* export company's info

mautic1

mautic2

#### Steps to test this PR:
1. Load up [this PR](https://m3.mautibox.com)
2. Create a company and assign some users to it
3. Go to Company menù and click to a company, it will take you to company view page.

### CODE REVIEW

**Comment by npracht** [comment]
Thanks @alfredoct96 this is an awesome contribution !!!!

File: app/bundles/LeadBundle/Views/Company/company.html.php [hunk 1]
```diff
@@ -0,0 +1,291 @@
+<?php
+
+/*
+ * @copyright   2014 Mautic Contributors. All rights reserved
```
Comment (GiuseppeIuculano) [inline app/bundles/LeadBundle/Views/Company/company.html.php:4]: Why copyright 2014 ? I think it should be 2020, but better to ask Mautic people :-)

**Comment by naveensirohi1** [comment]
Tried to test the info, found the following:
1. In the exported file the Name of the company does not appear, it shows the email. I blv the 
Screen Shot 2020-05-21 at 6 43 14 PM

Tests Done:
1. Created a company:
Companies_Info
2. Created a contact and added the contact to the company,
Screen Shot 2020-05-21 at 6 40 23 PM
3. Verified the following:
Added view page for companies.
It includes:
- Basic company's info
- Button to export company's info - Exports the information in .csv format. 
- Graph with sum of engagement and points of each company's contact
- Table with pagination of all contacts of company

Screen Shot 2020-05-21 at 7 03 18 PM
Screen Shot 2020-05-21 at 6 41 01 PM

@alfredoct96 -> How does point scenario work?

File: app/bundles/LeadBundle/Views/Company/list_rows_contacts.html.php [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+<?php
+
+$baseUrl = $view['router']->path(
+    'mautic_company_contacts_list',
+    [
+        'objectId' => $company->getId(),
+    ]
+);
+
+$customButtons = [
```
Comment (kuzmany) [inline app/bundles/LeadBundle/Views/Company/list_rows_contacts.html.php]: These lines looks like unnecessary

**Review by kuzmany — CHANGES_REQUESTED** [review]

File: app/bundles/LeadBundle/Views/Company/list_rows_contacts.html.php (hunk 2 above)
Comment (alfredoct96) [reply app/bundles/LeadBundle/Views/Company/list_rows_contacts.html.php]: Thanks! I removed the lines

**Pull request closed without merging**